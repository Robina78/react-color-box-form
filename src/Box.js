import React from "react";
import './Box.css'

const Box = ({id, handleRemove, width=5, height=5, backgroundColor="blue"}) => {
    const remove = () => handleRemove(id);
    return (
        <div className="Box">
           <div 
             style={{
                height: `${height}em`,
                width: `${width}em`,
                backgroundColor
             }}
            >
            <button onClick={remove}>Remove The Box!</button>
            </div>               
        </div>
    );
}

export default Box;