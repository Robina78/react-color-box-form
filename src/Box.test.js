import { render } from "@testing-library/react";
import React from "react";
import Box from './Box';

it("render without crashing", () => {
    render(<Box />);
});

it("matches snapshot", () => {
    const { asFragment } = render(<Box />);
    expect(asFragment()).toMatchSnapshot();
})