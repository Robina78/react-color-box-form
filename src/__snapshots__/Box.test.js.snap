// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <div
      style="height: 5em; width: 5em; background-color: blue;"
    >
      <button>
        Remove The Box!
      </button>
    </div>
  </div>
</DocumentFragment>
`;
